shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1343,26 +1333,24 @@ - (void)setMarkedText:(NSString*)markedText selectedRange:(NSRange)markedSelecte
     markedText = @"";
   }
 
-  if (markedTextRange.length > 0) {
-    // Replace text in the marked range with the new text.
-    [self replaceRangeLocal:markedTextRange withText:markedText];
-    actualReplacedRange = markedTextRange;
-    markedTextRange.length = markedText.length;
-  } else {
-    // Replace text in the selected range with the new text.
-    actualReplacedRange = selectedRange;
-    [self replaceRangeLocal:selectedRange withText:markedText];
-    markedTextRange = NSMakeRange(selectedRange.location, markedText.length);
-  }
+  const FlutterTextRange* currentMarkedTextRange = (FlutterTextRange*)self.markedTextRange;
+  const NSRange& actualReplacedRange = currentMarkedTextRange && !currentMarkedTextRange.isEmpty
+                                           ? currentMarkedTextRange.range
+                                           : _selectedTextRange.range;
+  // No need to call replaceRangeLocal as this method always adjusts the
+  // selected/marked text ranges anyways.
```

Comment:
I am a bit lost here. Can you explain more why we had this before, and we removed it now?

Reply:
`replaceRangeLocal` also changes the selected text range and the marked text range, based on the original selected text range and the marked text range and the text replacement. It's not needed in this method because with this method the caller always have to specify a new selected range and marked range anyways.

Reply:
Is the the same reason we removed the `markedText` state?

Reply:
`markedText`  was never used.